subiquity/server/controllers/oem.py

```diff
@@ -196,13 +193,16 @@ async def ensure_no_kernel_conflict(self) -> None:
 
         await self.kernel_configured_event.wait()
 
-        if self.model.metapkgs and kernel_model.explicitly_requested:
-            # TODO
-            # This should be a dialog or something rather than the content of
-            # an exception, really. But this is a simple way to print out
-            # something in autoinstall.
-            msg = _(
-                """\
+        if self.model.metapkgs:
+            for metapkg in self.model.metapkgs:
+                if kernel_model.metapkg_name == metapkg.name:
```

Comment:
Again here I think kernel_model.metapkg_name is something like linux-image-generic or linux-image-oem-22.04d, and metapkg.name is oem-*-meta

Reply:
while true, the way to resolve that is to walk the dependency chain

Reply:
per chat elsewhere, I do think this part is wrong and will revisit it